web.rb

```diff
@@ -116,3 +116,41 @@ def authenticate!
   status 200
   return "Charge successfully created"
 end
+
+# This endpoint responds to webhooks sent by Stripe. To use it, you'll need
+# to add its URL (https://{your-app-name}.herokuapp.com/stripe-webhook)
+# in the webhook settings section of the Dashboard.
+# https://dashboard.stripe.com/account/webhooks
+post '/stripe-webhook' do
+  json = JSON.parse(request.body.read)
+
+  # Retrieving the event from Stripe guarantees its authenticity
+  event = Stripe::Event.retrieve(json["id"])
+  source = event.data.object
+
+  # For sources that require additional user action from your customer
+  # (e.g. authorizing the payment with their bank), you should use webhooks
+  # to create a charge after the source becomes chargeable.
+  # For more information, see https://stripe.com/docs/sources#best-practices
+  if event.type == 'source.chargeable' && source.flow != 'none'
```

Comment:
I think we should have an explicit whitelist of type supported by the webhook endpoint here instead of a logic that might not remain valid.

How about defining:
```
# List of payment methods you support that rely on webhooks to create Charges.
WEBHOOK_CHARGE_CREATION_TYPES=['ideal', 'sofort']
```
?